### ISSUE TITLE
🐛 Dictionary migration doesn't check program or entity level exceptions

### ISSUE THREAD

**Issue by hknahal** [icgc-argo/argo-clinical#991]
## Describe the bug
If program or entity level exceptions are deleted, and then migrate to new dictionary, it does not trigger system to check program or entity level exceptions.

## Steps To Reproduce
1. Enabled program level exception for specimen schema to allow `tumour_grading_system` and `tumour_grade` fields to be `Unknown`.
2. Submitted specimen clinical file with `tumour_grading_system` = `Unknown` and `tumour_grade` = `Unknown`. Validated and signed-off on clinical data.
3. Deleted program level exception for specimen schema (no longer allows `tumour_grading_system` and `tumour_grade` to be `Unknown`).
4. Migrated to new dictionary. But no errors reported in `Submitted Data` view for previously submitted `tumour_grading_system` and `tumour_grade` fields if they are submitted as `Unknown`.

It appears dictionary migration doesn't trigger the system to check the program/entity level exceptions when re-validating.

## Expected behaviour
After dictionary migration, expected error to be reported in `Submitted Data` view for fields that are submitted as `Unknown` but have no exceptions applied anymore.

**Comment by joneubank** [comment]
@hknahal Question: what is the expected behaviour when an exception is removed? It's possible we need the system to re-evaluate all the data at that point instead of waiting for new submission or migration.

**Comment by hknahal** [comment]
@joneubank If an exception is deleted, the submitted data should be re-validated. For example, after Step #3 above (when exceptions for `tumour_grading_system` and `tumour_grade` are deleted), any fields that still have `tumour_grading_system` = `Unknown` or `tumour_grade` = `Unknown` should become invalid.

**Comment by joneubank** [comment]
I agree. To my knowledge, this is not done today?

I think we need to track two desired features:
1. re-validate after exceptions are modified
2. include exceptions in validations during migration (all validations should account for migrations)

**Comment by hknahal** [comment]
No, it's not done yet. I agree, the two features you mention would resolve this issue.

**Comment by Buwujiu** [comment]
@joneubank  @hknahal Do we want to release exceptions now or wait until we implement these two features?

**Comment by joneubank** [comment]
I would release and give us the ability to accept data with exceptions. Removing exceptions is a use case we don't have immediate expectation to need.

**Comment by hknahal** [comment]
I think we should go ahead and release exceptions instead of waiting. This won't block any of the programs.

**Comment by Buwujiu** [comment]
Same thoughts here! I will make a plan to track these two issues, and we shall be able to prepare for the release!
